### PR TITLE
[Stable] Use latest terra release for stable branch CI

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

This commit switches the version of terra we use for testing on the
stable branch. Currently we test everything from git on the master
branch for CI including the stable branches which is just carried over
from master. However for the stable branch it's more desirable to test
against terra from releases. We track master on aer's master to ensure
the next release will be able to accommodate any breaking changes or use
new features, but for the stable we need to ensure it works against
what's already been released for terra and remains stable and usable for
our end users. Testing against master both masks this and introduces
more noise into CI because it will potentially introduce failures
because of changes in terra development. This switches to use terra's
release to avoid this and make CI more stable for the stable branch.

### Details and comments

### CODE REVIEW

**Comment by mtreinish** [comment]
Superseded by: https://github.com/Qiskit/qiskit-aer/pull/636

File: azure-pipelines.yml [hunk 1]
```diff
@@ -138,14 +138,14 @@ stages:
               source activate qiskit-aer-$(Build.BuildNumber)
               conda install -c conda-forge --yes --quiet --name qiskit-aer-$(Build.BuildNumber) python=$(python.version) osqp
               pip install -r requirements-dev.txt
-              pip install git+https://github.com/Qiskit/qiskit-terra.git
+              pip install qiskit-terra
```
Comment (atilag) [inline azure-pipelines.yml]: Shouldn't we put conditionals depending on the branch here as well?

Comment (mtreinish) [reply azure-pipelines.yml]: oops, yep I completely forgot about  azure too. I'll update this now

Comment (mtreinish) [reply azure-pipelines.yml]: Done in: https://github.com/Qiskit/qiskit-aer/pull/633/commits/801d6293429e1ea9865da2a4ba651d240693c453

**Comment by atilag** [comment]
Closing it, because we are going to merge from `master` (which already include these changes)

**Pull request closed without merging**